spacy/cli/_util.py

```diff
@@ -221,24 +221,45 @@ def validate_project_commands(config: Dict[str, Any]) -> None:
 
     config (Dict[str, Any]): The loaded config.
     """
+
+    def verify_workflow_step(workflow_name: str, step: str) -> None:
+        if step not in command_names:
+            msg.fail(
+                f"Unknown command specified in workflow '{workflow_name}': {step}",
+                f"Workflows can only refer to commands defined in the 'commands' "
+                f"section of the {PROJECT_FILE}.",
+                exits=1,
+            )
+
     command_names = [cmd["name"] for cmd in config.get("commands", [])]
     workflows = config.get("workflows", {})
     duplicates = set([cmd for cmd in command_names if command_names.count(cmd) > 1])
     if duplicates:
         err = f"Duplicate commands defined in {PROJECT_FILE}: {', '.join(duplicates)}"
         msg.fail(err, exits=1)
-    for workflow_name, workflow_steps in workflows.items():
+    for workflow_name, workflow_step_or_lists in workflows.items():
```

Comment:
Let's find a shorter name for `workflow_step_or_lists`, maybe just `commands`? Then what is currently `command_names` could be `all_commands` or such.